### ISSUE TITLE
Caption under images

### ISSUE THREAD

**Issue by jdumas** [gnab/remark#408]
Hi,

I'm often inserting images into my slides using html to easily control their size:

`<img src="my_figure.png" width="40%">`

However, I am struggling to easily put a caption below them. E.g. let's say I have 3 pictures in my slide with a width of `30%`. How do I put a small caption below that is centered wrt each image?

**Comment by tripu** [comment]
@jdumas, have you tried with [flexbox](https://www.w3.org/TR/css-flexbox/)?

**Comment by jdumas** [comment]
I'm a bit of a css noob. Could you provide an example of what you suggest?

**Comment by tripu** [comment]
> Could you provide an example of what you suggest?

Apart from the spec linked above: a [tutorial](https://css-tricks.com/snippets/css/a-guide-to-flexbox/), some [visual examples to interact with](https://codepen.io/enxaneta/pen/adLPwv).

**Comment by jdumas** [comment]
Thanks for the resources. I'll try to see if it can solve my caption problem.

**Comment by tripu** [comment]
@jdumas, [this was my suggestion](https://codepen.io/tripu/pen/YNZrmW). Nothing to do with remark specifically though; it's vanilla HTML &amp; CSS. I'm not sure whether/how Markdown will swallow this piece of HTML, but in theory it should be okay to embed this in any Markdown source document.

**Comment by jdumas** [comment]
Yes we can mix html with markdown. But with remark we can also define our own css class in the html header, and use them in the markdown source (e.g. `.foo[My text]`). So I was hoping we could define some magic bongo to avoid having to layout everything in pure html (since once I open a `<div>` in the markdown source, I cannot put markdown inside, I have to do it in pure html, which is a pain).

**Comment by tripu** [comment]
@jdumas, like [this](https://codepen.io/tripu/pen/JEWMdJ)?

**Comment by jdumas** [comment]
Yes! Nice example =).
Also, I think we can get rid of the hardcoded `33.3%` width by using `justify-content: space-around;` in `.special-list ul`

**Comment by tripu** [comment]
> I think we can get rid of the `hardcoded 33.3%` width by using `justify-content: space-around;` in `.special-list ul`.

You're right. I updated the pen with `justify-content: space-between;`.

**Comment by jdumas** [comment]
Actually I do notice something a bit weird when I try on my presentation. Starting from the boilerplate html code and running this MWE:

```html
<!DOCTYPE html>
<html>
  <head>
    <title>Title</title>
    <meta charset="utf-8">
    <style>
      @import url(https://fonts.googleapis.com/css?family=Yanone+Kaffeesatz);
      @import url(https://fonts.googleapis.com/css?family=Droid+Serif:400,700,400italic);
      @import url(https://fonts.googleapis.com/css?family=Ubuntu+Mono:400,700,400italic);

      body { font-family: 'Droid Serif'; }
      h1, h2, h3 {
        font-family: 'Yanone Kaffeesatz';
        font-weight: normal;
      }

      .special-list ul {
        text-align: center;
        display: flex;
        justify-content: space-between;
      }

      .special-list li {
        list-style-type: none;
      }

      .caption {
        display: block;
      }

      .remark-code, .remark-inline-code { font-family: 'Ubuntu Mono'; }
    </style>
  </head>
  <body>
    <textarea id="source">

class: center, middle

# Title

---

# Agenda

1. Introduction
2. Deep-dive
3. ...

---
# with flexbox layout

.special-list[
* ![Mizaru](https://upload.wikimedia.org/wikipedia/commons/6/6e/Emoji_u1f648.svg) .caption[*Mizaru* (see no evil)]
* ![Kikazaru](https://upload.wikimedia.org/wikipedia/commons/5/56/Emoji_u1f649.svg) .caption[*Kikazaru* (hear no evil)]
* ![Iwazaru](https://upload.wikimedia.org/wikipedia/commons/2/2c/Emoji_u1f64a.svg) .caption[*Iwazaru* (speak no evil)]
]

---

# Introduction

    </textarea>
    <script src="https://gnab.github.io/remark/downloads/remark-latest.min.js">
    </script>
    <script>
      var slideshow = remark.create();
    </script>
  </body>
</html>
```

I get a weird padding of 40 pixels created by the `ul` tag. Any idea what's causing this discrepancy? Different version of remark.js?

**Comment by jdumas** [comment]
Also, if you have any tips on how to control vertical alignment nicely through markdown, using this special-list, that would be nice! I am struggling with the `align-self: flex-end;` or `align-items:` properties, to no avail =/

**Comment by tripu** [comment]
> Any idea what's causing this discrepancy?

@jdumas: it's difficult to tell without looking at the actual document.

Your browser's [default style sheets](https://meiert.com/en/blog/20070922/user-agent-style-sheets/) might be setting those margins/paddings (in which case, you can use a tool like [Normalize.css](https://necolas.github.io/normalize.css/) to get consistent results across all browsers). Or you could have other style sheets of yours interfering.

Feel free to fork [my original pen](https://codepen.io/tripu/pen/JEWMdJ) to show us the issue.

**Comment by tripu** [comment]
> if you have any tips on how to control vertical alignment nicely through markdown, using this special-list, that would be nice!

@jdumas: I have modified [my pen](https://codepen.io/tripu/pen/JEWMdJ) to show how to align items vertically. I've added these two properties:

```css
.special-list ul {
  ⋮
  align-items: center;
}

.special-list li {
  ⋮
  flex-basis: 33.33%;
}
```

**Comment by jdumas** [comment]
Thanks for the baseline tip!

As for the padding, I have the same issue on both firefox & chromium. I cannot reproduce in the codepen since the only thing I have done is copy you example into the boilerplate code. Could it be an update in remark.js that introduces this padding? How do I see which version of remarkjs is used in your codepen?

**Comment by tripu** [comment]
My pen is linking to [`remarkjs.com/downloads/remark-latest.min.js`](https://remarkjs.com/downloads/remark-latest.min.js) (click on the gears icon in the JS panel).

You can use your browsers' &ldquo;developer tools&rdquo; to inspect those margins and paddings, and find out where exactly they come from.

**Comment by jdumas** [comment]
Yes you are right. Sorry for the inconvenience. I had a `-webkit-padding-start: 40px;` laying around somewhere. Wtf. Normalize.css did not reset this attribute, but Meyer's CSS Reset did (well, they are different after all). I am still unsure which one I should go for.

**Comment by pat-s** [comment]
You can try this:

```html
<figure>
    <img src='/path/to/image.png' />
      <font size="2">
        <figcaption> some caption here
        </figcaption>
      </font>
</figure>
```

Caption is left-aligned. However, it should be easy to center the caption searching the web

**Comment by 1-2-3** [comment]
I made a template for remarkjs that contains some themes, layouts, and components. I hope it works for you.
[demo](https://1-2-3.github.io/remark-it/index-en_US.html) | [source](https://github.com/1-2-3/remark-it)
![](https://raw.githubusercontent.com/1-2-3/remark-it/master/screenshots/111.png?raw=true)
![](https://raw.githubusercontent.com/1-2-3/remark-it/master/screenshots/222.png?raw=true)
